tests/functional/test_main.py

```diff
@@ -0,0 +1,161 @@
+from cgi import test
+from http import HTTPStatus
+from flask_login import login_user, current_user
+from flask import session
+
+
+def test_root(test_app, test_db):
+
+    with test_app.test_client() as test_client:
+        response = test_client.get("/")
+        assert response.status_code == HTTPStatus.OK
+
+
+def test_login_get(test_app):
+
+    with test_app.test_client() as test_client:
+        response = test_client.get("login")
+        assert response.status_code == HTTPStatus.OK
+
+
+def test_login_post(test_app, persisted_user):
+
+    with test_app.test_client() as test_client:
```

Comment:
Use pytest-flask client or client-class instead to save some boilerplate

https://pytest-flask.readthedocs.io/en/latest/features.html#client-application-test-client
https://pytest-flask.readthedocs.io/en/latest/features.html#client-class-application-test-client-for-class-based-tests